source/client/stream_decoder.cc

```diff
@@ -87,11 +87,15 @@ void StreamDecoder::onPoolReady(Envoy::Http::StreamEncoder& encoder,
     // we take the risk of erroneously reporting that we did send all the bytes, instead of always
     // reporting 0 bytes.
     stream_info_.addBytesReceived(request_body_size_);
-    // TODO(oschaaf): We can probably get away without allocating/copying here if we pregenerate
-    // potential request-body candidates up front. Revisit this when we have non-uniform request
-    // distributions and on-the-fly reconfiguration in place.
-    std::string body(request_body_size_, 'a');
-    Envoy::Buffer::OwnedImpl body_buffer(body);
+    // Revisit this when we have non-uniform request distributions and on-the-fly reconfiguration in
```

Comment:
On the fly reconfiguration is something we are after, so this is probably a temporary performance increase. How are we going to deal with the performance regression that will come with switching this back to copying?

Are we going to make this optional and only copy on reconfiguration? Do you have other plans? I am worried about users starting to depend on this newly increased performance if we are going to end up reverting it.

Reply:
I guess that what is missing is an abstraction to source the actual contents we are sending. We only have a content-length concept, which will work well even when that requested content-length changes dynamically. But today there's no way in place to change the actual content that we are sending. The series of 'a' characters we use today probably is too simple / unfit for certain purposes (for one that is highly compressible), maybe we should take this to slack for a bit and discuss where we want to take this?

Reply:
Discussed offline and the questions were answered, this won't prevent us from implementing any dynamic body functionality.